### ISSUE TITLE
Recursive call from typeof in output bundle

### ISSUE THREAD

**Issue by jcubic** [rollup/rollup-plugin-babel#252]
I finally was able to make rollup + babel to include regenerator and when I've included output library in browser it give me this error:

```
lib.js:16 Uncaught RangeError: Maximum call stack size exceeded
    at _typeof2 (lib.js:16)
    at _typeof (lib.js:29)
    at _typeof2 (lib.js:17)
    at _typeof (lib.js:29)
    at _typeof2 (lib.js:17)
    at _typeof (lib.js:29)
    at _typeof2 (lib.js:17)
    at _typeof (lib.js:29)
    at _typeof2 (lib.js:17)
    at _typeof (lib.js:29)
```

the lines look like this:
```javascript
  var _typeof_1$$1 = createCommonjsModule(function (module) {
    function _typeof2(obj) {
      if (typeof Symbol === "function" && _typeof_1$$1(Symbol.iterator) === "symbol") {
        _typeof2 = function _typeof2(obj) {
          return _typeof_1$$1(obj);
        };
      } else {
         _typeof2 = function _typeof2(obj) {
          return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : _typeof_1$$1(obj);
        };
      }
      return _typeof2(obj);
    }
    function _typeof(obj) {
      if (typeof Symbol === "function" && _typeof2(Symbol.iterator) === "symbol") {
        module.exports = _typeof = function _typeof(obj) {
          return _typeof2(obj);
        };
      } else {
        module.exports = _typeof = function _typeof(obj) {
          return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : _typeof2(obj);
        };
      }
      return _typeof(obj);
    }
    module.exports = _typeof;
  });
```
my rollup config look like this:
```javascript
import commonjs from "rollup-plugin-commonjs";
import nodeResolve from "rollup-plugin-node-resolve";
import babel from "rollup-plugin-babel";

export default {
    input: "src/main.js",
    output: {
        name: "lib",
        file: "dist/bundle.js",
        format: "iife"
    },
    plugins: [
        commonjs({
            include: "node_modules/**"
        }),
        babel({
            "babelrc": false,
            "runtimeHelpers": true,
            "plugins": [
                "syntax-async-functions",
                "@babel/plugin-external-helpers",
                "@babel/plugin-transform-async-to-generator",
                ["@babel/plugin-transform-runtime", {
                    "helpers": true,
                    "regenerator": true
                }]
            ],
            "presets": [
                "@babel/preset-env"
            ]
        }),
        nodeResolve({
          jsnext: true,
          main: false
        })
    ]
};
```

**Comment by jcubic** [comment]
I think it's babel error again as in babel/babel/issues/3836

**Comment by jcubic** [comment]
The work around is to use replace plugin:
```javascript
        replace({
            delimiters: ["", ""],
            "_typeof2(Symbol.iterator)": "typeof Symbol.iterator",
            "_typeof2(obj);": "typeof obj;"
        })
```

**Comment by jcubic** [comment]
it give warning:

```
(!) Circular dependency: node_modules/@babel/runtime/helpers/typeof.js -> node_modules/@babel/runtime/helpers/typeof.js
```

but it works

**Comment by Andarist** [comment]
I doubt this is causing your problem, but you should remove `"@babel/plugin-external-helpers"`. As mentioned in another thread - without the actual repository reproducing the issue it's hard to help. Could you provide one?

**Comment by jcubic** [comment]
Here is my repo with replace fix https://github.com/jcubic/open-source-library

**Comment by jcubic** [comment]
PR resolved the issue jcubic/open-source-library#1

**Comment by pkhodaveissi** [comment]
I still have this warning though:
```(!) Circular dependency: node_modules/@babel/runtime/helpers/typeof.js -> node_modules/@babel/runtime/helpers/typeof.js```
but works at the end
also my babel looks like this:
```
babel({
      babelrc: false,
      runtimeHelpers: true,
      plugins: [
        "@babel/plugin-transform-async-to-generator",
        ["@babel/plugin-transform-runtime", { useESModules: true }]
      ],
      presets: ["@babel/preset-env"],
      exclude: "node_modules/**"
    })```
